x/distribution/types/keys.go

```diff
@@ -52,27 +54,52 @@ var (
 	ValidatorOutstandingRewardsPrefix    = collections.NewPrefix(2) // key for outstanding rewards
 	DelegatorWithdrawAddrPrefix          = collections.NewPrefix(3) // key for delegator withdraw address
 	DelegatorStartingInfoPrefix          = collections.NewPrefix(4) // key for delegator starting info
-	ValidatorHistoricalRewardsPrefix     = []byte{0x05}             // key for historical validators rewards / stake
+	ValidatorHistoricalRewardsPrefix     = collections.NewPrefix(5) // key for historical validators rewards / stake
 	ValidatorCurrentRewardsPrefix        = collections.NewPrefix(6) // key for current validator rewards
 	ValidatorAccumulatedCommissionPrefix = collections.NewPrefix(7) // key for accumulated validator commission
 	ValidatorSlashEventPrefix            = []byte{0x08}             // key for validator slash fraction
 	ParamsKey                            = collections.NewPrefix(9) // key for distribution module params
 )
 
-// GetValidatorHistoricalRewardsAddressPeriod creates the address & period from a validator's historical rewards key.
-func GetValidatorHistoricalRewardsAddressPeriod(key []byte) (valAddr sdk.ValAddress, period uint64) {
-	// key is in the format:
-	// 0x05<valAddrLen (1 Byte)><valAddr_Bytes><period_Bytes>
-	kv.AssertKeyAtLeastLength(key, 2)
-	valAddrLen := int(key[1])
-	kv.AssertKeyAtLeastLength(key, 3+valAddrLen)
-	valAddr = sdk.ValAddress(key[2 : 2+valAddrLen])
-	b := key[2+valAddrLen:]
-	kv.AssertKeyLength(b, 8)
-	period = binary.LittleEndian.Uint64(b)
-	return
+// LEUint64Key is a collections KeyCodec that encodes uint64 using little endian.
+// NOTE: it MUST NOT be used by other modules, distribution relies on this only for
+// state backwards compatibility.
+var LEUint64Key collcodec.KeyCodec[uint64] = leUint64Key{}
```

Comment:
If it shouldn't be imported, why not put that in an internal package? I guess it is fine too like that, but possibly, we should add a deprecated comment.